sycl/include/CL/sycl/detail/property_helper.hpp

```diff
@@ -0,0 +1,82 @@
+//==--------- property_helper.hpp --- SYCL property helper -----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/common.hpp>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+
+namespace detail {
+
+// All properties are split here to simple and complex. A simple property is one
+// which has no data stored in it. A complex property is one which has data
+// stored in it and usually provides and access to it.
+// For simple property we just store a bool which indicates if a property is
+// set or not. For complex properties we store a pointer to the base class
+// because we do not know the size of such properties beforehand.
```

Comment:
Why not naming them accordingly?
Such as `DataLessPropKind` and `PropWithDataKind` or something like that?

Reply:
Or just `PropKind` in both cases but declared inside `DataLessPropertyBase` and `PropertyWithDataBase` classes?

Reply:
Don't like second version. Will rename to DataLessPropKind and PropWithDataKind.